awx/main/models/inventory.py

```diff
@@ -48,6 +48,9 @@ class Inventory(CommonModelNameNotUnique, ResourceMixin):
     an inventory source contains lists and hosts.
     '''
 
+    FIELDS_TO_PRESERVE_AT_COPY = [
+        'hosts', 'groups', 'insights_credential', 'organization', 'instance_groups'
+    ]
```

Comment:
`host_filter`, because someone might be copying a smart inventory.

Yes, you could just add that to the list, but there's a broader point that lists like these are hard to maintain and commit us to a large amount of future iteration. I would like to know if there's some kind of rule that we can use to more easily generate the list.

```python
In [1]: [f.name for f in Inventory._meta.fields]
Out[1]: 
[u'id',
 'created',
 'modified',
 'description',
 'created_by',
 'modified_by',
 'name',
 'organization',
 'variables',
 'has_active_failures',
 'total_hosts',
 'hosts_with_active_failures',
 'total_groups',
 'groups_with_active_failures',
 'has_inventory_sources',
 'total_inventory_sources',
 'inventory_sources_with_failures',
 'kind',
 'host_filter',
 'admin_role',
 'update_role',
 'adhoc_role',
 'use_role',
 'read_role',
 'insights_credential',
 'pending_deletion']
```

This list would need to be processed, but you could have a global method that does most of the important processing. Start with removing `id`, the role fields, etc. But then others require complete re-thinking. `insights_credential` has RBAC implications. `pending_deletion` should never be True to begin with if the action is allowed.

Reply:
It is not necessary to include `host_filter` because that field is part of the create POST body, so will be auto-fetched.